include/hip/devicelib/macros.hh

```diff
@@ -52,4 +52,7 @@
 typedef _Float16 api_half;
 typedef _Float16 api_half2 __attribute__((ext_vector_type(2)));
 
+typedef unsigned int uint;
+typedef unsigned long ulong;
```

Comment:
Why is this necessary on macOS? Does it still work for other platforms or will it report duplicate identifier?

Reply:
It might cause troubles. On my system (Ubuntu 22.04) <stdlib.h> includes <sys/types.h> which defines typedefs for u{short,int,long}.

/usr/include/x86_64-linux-gnu/sys/types.h:
```
...
#ifdef __USE_MISC
/* Old compatibility names for C types.  */
typedef unsigned long int ulong;
typedef unsigned short int ushort;
typedef unsigned int uint;
#endif
...
```

Reply:
Actually, it's not a problem as long as the duplicate typedefs aliases the same type.